### PR TITLE
Compute correct normals for PolygonGeometry using perPositionHeight

### PR DESCRIPTION
**Pull request by hpinkos**
We weren't actually computing the normals when `perPositonHeight === true`, instead we were just using the ellipsoid surface normal.
This uses `GeometryPipeline.computeNormal` to compute the normals for the top polygon (the perPositionHeight one) 
This also fixes the normals on the walls of an extruded perPositionHeight polygon if one of the positions has the same height as the extrusion

### CODE REVIEW

**Comment by hpinkos** [comment]
I tested with this sandcastle demo:
``` javascript
var viewer = new Cesium.Viewer('cesiumContainer');
var scene = viewer.scene;

var pos1 = [new Cesium.Cartesian3(1333465.894653764, -4654487.507295778, 4138588.25016027),
            new Cesium.Cartesian3(1333485.211963876, -4654510.505548239, 4138557.5850382405),
            new Cesium.Cartesian3(1333385.4137239386, -4654258.530765127, 4138343.625877602)];

var pos2 = [new Cesium.Cartesian3(1333502.5341302191, -4654468.273531829, 4138599.2196902917),
            new Cesium.Cartesian3(1333465.5540792884, -4654486.318514235, 4138587.186018895),
            new Cesium.Cartesian3(1333403.0838373178, -4654216.321704011, 4138385.1235247967)];

var pos3 = [new Cesium.Cartesian3(1333502.4040933265, -4654467.819649983, 4138598.813393836),
            new Cesium.Cartesian3(1333522.047929563, -4654490.698890668, 4138567.922178115),
            new Cesium.Cartesian3(1333459.0591281147, -4654218.933622686, 4138364.2905654362)];

var pos4 = [new Cesium.Cartesian3(1333485.211963876, -4654510.505548239, 4138557.5850382405),
            new Cesium.Cartesian3(1333441.3994441305, -4654261.147368878, 4138322.784348336),
            new Cesium.Cartesian3(1333521.9333286814, -4654490.298890729, 4138567.564118971)];

var geometry = Cesium.PolygonGeometry.createGeometry(Cesium.PolygonGeometry.fromPositions({
    positions: pos1,
    extrudedHeight: 56,
    vertexFormat: Cesium.VertexFormat.POSITION_AND_NORMAL,
    perPositionHeight: true,
    closeBottom: false
}));

scene.primitives.add(new Cesium.Primitive({
    geometryInstances : new Cesium.GeometryInstance({
        geometry : geometry,
        attributes: {
            color: Cesium.ColorGeometryInstanceAttribute.fromColor(Cesium.Color.RED)
        }
    }),
    appearance : new Cesium.PerInstanceColorAppearance({
        closed : true,
        translucent : false
    }),
    asynchronous: false
}));

scene.primitives.add(Cesium.createTangentSpaceDebugPrimitive({
     geometry : geometry,
     length : 1000.0,
}));

scene.camera.setView({
    destination: new Cesium.Cartesian3(1333502.4040933265, -4654467.819649983, 4138598.813393836)
});
```

File: Specs/Core/PolygonGeometrySpec.js [hunk 1]
```diff
@@ -596,6 +598,35 @@ defineSuite([
...
+        var expectedNormals = geometry.attributes.normal.values;
+
+        var notEqualCount = 0;
+        for (var i = 0; i < expectedNormals.length; i++) {
+            if (!CesiumMath.equalsEpsilon(normals[i], expectedNormals[i], CesiumMath.EPSILON6)) {
+                notEqualCount++;
+            }
+        }
+
+        //Exactly 2 normals will be different due to weird triangles on the walls of the extrusion
```
Comment (mramato) [inline Specs/Core/PolygonGeometrySpec.js]: We should probably write up a separate issue for this and add a link to it here.

**Comment by mramato** [comment]
@bagnell this is all you when you have time.

Thanks @hpinkos!

**Comment by mramato** [comment]
I was hoping this would also fix #2399, but that does not appear to be the case.

File: Specs/Core/PolygonGeometrySpec.js (hunk 1 above)
Comment (hpinkos) [inline Specs/Core/PolygonGeometrySpec.js]: Created #3885

File: Source/Core/PolygonGeometryLibrary.js [hunk 2]
```diff
@@ -307,6 +308,7 @@ define([
                 indices : indices,
                 primitiveType : PrimitiveType.TRIANGLES
             });
+            return GeometryPipeline.computeNormal(geometry);
```
Comment (bagnell) [inline Source/Core/PolygonGeometryLibrary.js]: Polygons with per position heights always have normals even if they don't need them?

**Comment by hpinkos** [comment]
@bagnell ready

**Pull request merged**